modules/salt/pages/large-scale.adoc

```diff
@@ -0,0 +1,177 @@
+For installations with more than 1000 clients per {productname} Server, adequate hardware sizing and parameter tuning must be performed.
+
+There is no hard maximum number of supported systems.
+Many factors can affect how many clients can reliably be used in a particular installation.
+Factors can include which features are used, and how the hardware and systems are configured.
+
+
+[WARNING]
+====
+Large installations require standard Salt clients.
+These instructions cannot be used in environments using traditional clients or Salt SSH minions.
+====
+
+
+
+== Hardware and Infrastructure
+
+Not all problems can be solved with better hardware, but choosing the right hardware is an absolute necessity for large scale deployments.
+
+
+The minimum requirements for the {productname} Server are:
+
+* Eight or more recent {x86_64} CPU cores.
+* 32{nbsp}GiB RAM.
+For installations with thousands of clients, use 64{nbsp}GB or more.
+* Fast I/O storage devices, such as locally-attached SSDs.
+For PostgreSQL data directories, we recommend locally-attached RAID-0 SSDs.
+
+If the {productname} Server is virtualized, enable the `elevator=noop` kernel command line option, for optimal I/O performance.
+You can check the current status with [command]``cat /sys/block/<DEVICE>/queue/scheduler``.
+This command will display a list of available schedulers with the currently active one in brackets.
+To change the scheduler before a reboot, use [command]``echo noop > /sys/block/<DEVICE>/queue/scheduler``.
+
+The minimum requirements for the {productname} Proxy are:
+
+* One {productname} Proxy per 500-1000 clients, depending on available network bandwidth.
+* Two or more recent {x86_64} CPU cores.
+* 16{nbsp}GiB RAM, and sufficient storage for caching.
+
+Clients should never be directly attached to the {productname} Server in production systems.
+
+In large scale installations, the {productname} Proxy is used primarily as a local cache for content between the server and clients.
+Using proxies in this way can substantially reduce download time for clients, and decrease Server egress bandwidth use.
+
+The number of clients per proxy will affect the download time.
+Always take network structure and available bandwidth into account.
+
+We recommend you estimate the download time of typical usage to determine how many clients to connnect to each proxy.
+To do this, you will need to estimate the number of package upgrades required in every patch cycle.
+You can use this formula to calculate the download time:
+
+----
+Size of updates * Number of clients / Theoretical download speed / 60
+----
+
+For example, the total time needed to transfer 400{nbsp}MB of upgrades through a physical link speed of 1{nbsp}GB/s to 3000 clients:
+
+----
+400 MB  * 3000 / 119 MB/s / 60 = 169 min
+----
+
+
+
+== Operation Recommendations
+
+This section contains a range of recommendations for large scale deployments.
+
+[NOTE]
+====
+Always start small and scale up gradually.
```

Comment:
@Loquacity I took the liberty to move this under the Operation Recommendations, as it's not really about hardware/infrastructure requirements.

Feel free to just revert commit f9241ca2  in case you want it back.